Kentico.Kontent.Delivery.Extensions.DependencyInjection/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -17,6 +19,28 @@ namespace Kentico.Kontent.Delivery.Extensions.DependencyInjection
     /// </summary>
     public static class ServiceCollectionExtensions
     {
+
+        /// <summary>
+        /// Registers a delegate that will be used to configure a named <see cref="IDeliveryClient"/> via <see cref="IDeliveryClientFactory"/>
+        /// </summary>
+        /// <param name="services">A <see cref="ServiceCollection"/> instance for registering and resolving dependencies.</param>
+        /// <param name="buildDeliveryClientFactory">A function that is provided with an instance of <see cref="IDeliveryClientFactoryBuilder"/>and expected to return a valid instance of <see cref="IDeliveryClientFactory"/>.</param>
+        /// <returns>The <paramref name="services"/> instance with <see cref="IDeliveryClientFactory"/> registered in it</returns>
+        public static IServiceCollection AddDeliveryClientFactory(this IServiceCollection services, Func<IDeliveryClientFactoryBuilder, IDeliveryClientFactory> buildDeliveryClientFactory)
+        {
+            if (buildDeliveryClientFactory == null)
+            {
+                throw new ArgumentNullException(nameof(buildDeliveryClientFactory), "The function for creating Delivery client factory is null.");
+            }
+
+            var factory = buildDeliveryClientFactory(new DeliveryClientFactoryBuilder());
+
+            // TODO 312 - TyAdd or Add with exception?
+            services.TryAddSingleton<IDeliveryClientFactory>(factory);
```

Comment:
I would throw an exception, as a user I would be surprised if I registered something and it silently failed

Reply:
Yes